### ISSUE TITLE
Activity 0205 str_replace_all

### ISSUE THREAD

**Issue by chycinsa** [STA518-Spring20/discussion#71]
I understand how to remove the "featuring", but I can't figure out how to remove everything before the featuring.

**Comment by dykesb** [comment]
We saw some of this back in Activity 2.2 (e.g., here is a response I gave in #40):
> `names_sep` asks for when to separate (i.e., what character index). So if you want to separate around the number value in the string, you would separate the string at 2 and 3 and set the middle name to `NA` (if you separate a string at 2 places you have 3 columns. For example, "splitat6and8" with `names_sep = c(6,8)` becomes "split" "at" "6and8". So I would need to have `names_to = c("first", NA, "values")` and the "at" column wouldn't show.
> 
> Try using `names_pattern = "(.)\\d(.*)"` instead. This pattern says "keep the first value "`(.)`", remove digit value "`\\d`", keep the next value and everything afterwards "`(.*)`". You would only need two `names_to` values using this method. If I wanted to also keep the digit value, I would put that in within parentheses and my `names_to` argument would need another value.

Since we aren't providing a column pattern, we do not need to worry about parentheses to say what to keep, the various `str_...()` functions handle this for us.  


**TL;DR**: `.*` will be super helpful when parsing strings when they aren't the same length.

**Comment by chycinsa** [comment]
How do I specify featuring. I tried `"//featuring.*"` but obviously that doesn't work.

**Comment by dykesb** [comment]
What do the `"//"` do?

**Comment by chycinsa** [comment]
I typed them in the wrong direction, I was using `\\featuring.*`

**Comment by dykesb** [comment]
Ok, so what are you trying to do with `"\\"`? 

The backslash `\` is used to escape special characters (e.g., to get `"This is a quoted sentence."` to print with the quotations, it would be `\"This is an quoted sentence.\"`). So what your current string is looking for is stuff that contains `"\"` (because `\\` would display as `\`), then `"featuring"`, then `.*` (everything after `"\featuring"`).
